doc/ideas/savefileformat.md

```diff
@@ -0,0 +1,446 @@
+
+
+| id  | type      | children | info         |
+| --- | --------- | -------- | ------------ |
+| 0   | condition | 0        | leaf of tree |
+| 1   | NOT       | 1        | node         |
+| 2   | OR        | 2        | node         |
+| 3   | AND       | 2        | node         |
+| 4   | XOR       | 2        | node         |
+
+
+```
+                    root
+                    and
+                   /    \
+                 or    player_has_units_of_type
+               /    \
+       time_passed  unit_is_type
+```
+
+serialisation/deserialisation:
+
+
+```
+                       +-----+
+                       |start|
+                       +--+--+
+                          |
+                          |
+                          v
+                      +---+----+     no    +---------------+
+         +----------->+is node?+---------->+write condition|
+         |            +---+----+           |   to stdout   |
+         |                |                +---------------+
+         |                |  yes
+         |                v
+         |           +----+------+
+recursion|           | write node|
+         |           | to stdout |
+         |           +----+------+
+         |                |
+         |                |
+         |                v
+         |      +---------+------------+
+         |      |recursive left        |
+         |      |recursive right       |
+         +------+or                    |
+                |single recursive (not)|
+                +----------------------+
+
+                        +-----+
+                        |start|
+                        +--+--+
+                           |
+                           |
+                           v               +--------------+
+                       +---+----+   no     |read condition|
+         +------------>+is node?+--------->+   return it  |
+         |             +----+---+          +--------------+
+         |                  |
+         |                  |yes
+recursion|                  v
+         |      +-----------+----------------+
+         |      |read node                   |
+         |      |left  node = read recursive |
+         |      |right node = read recursive |
+         +------+or                          |
+                |single node = read recursive|
+                |return node                 |
+                +----------------------------+
+
+```
+
+
+### 11. Crypto
+
+everything from byte 826-end will be encrypted with aes.
+
+Key distribution:
+
+- encryption key will be randomly generated
+- server writes savegame unencrypted to disk
+- server encrypts everything after the last header byte with aes until file ending
+- server writes crypto IV in header field
+- generate team-count-1 times a random 256Bit(encryption key size) pseudo key
+- key team N(the one without pseudo key): team1 xor team2 xor .. teamN-1 xor original encryption key and destroys the original encryption key
```

Comment:
this idea is nice, i'll show it to some of my crypto-experienced friends cause I can't tell if it has major loopholes.